lib/rules/no-mocha-arrows.js

```diff
@@ -16,21 +15,31 @@ module.exports = {
     create(context) {
         const sourceCode = context.getSourceCode();
 
+        // eslint-disable-next-line max-statements
         function formatFunctionHead(fn) {
-            const paramsLeftParen = sourceCode.getFirstToken(fn);
-            const paramsRightParen = sourceCode.getTokenBefore(sourceCode.getTokenBefore(fn.body));
-            let paramsFullText = sourceCode.text.slice(paramsLeftParen.range[0], paramsRightParen.range[1]);
+            const arrow = sourceCode.getTokenBefore(fn.body);
+            let firstToken = sourceCode.getFirstToken(fn);
+            const beforeArrowToken = sourceCode.getTokenBefore(arrow);
+            let paramsFullText;
+            let params = sourceCode.text.slice(firstToken.range[0], beforeArrowToken.range[1]).trim();
             let functionKeyword = 'function';
+            const beforeArrowComment = sourceCode.text.slice(beforeArrowToken.range[1], arrow.range[0]).trim();
+            const afterArrowComment = sourceCode.text.slice(arrow.range[1], fn.body.range[0]).trim();
```

Comment:
For better readability I would suggest to extract the repeated pattern of `sourceCode.text.slice(start, end).trim()` to a function.

Reply:
Is
```js
sliceAndTrim( sourceCode.text, beforeArrowToken.range[1], arrow.range[0] );
```
much more concise or legible?

Reply:
I think `sourceCode.text` doesn’t need to be a parameter when we define the new function in the scope of the `create` function. Maybe a better name would be something like `extractSourceTextByRange()`. WDYT?